src/daemon/delayed_shutdown_timer.cpp

```diff
@@ -48,19 +66,40 @@ void mp::DelayedShutdownTimer::start(std::chrono::milliseconds delay)
                  fmt::format("Shutdown request delayed for {} minute{}",
                              std::chrono::duration_cast<std::chrono::minutes>(delay).count(),
                              delay > std::chrono::minutes(1) ? "s" : ""));
-        QObject::connect(&shutdown_timer, &QTimer::timeout, [this]() { shutdown_instance(); });
+        ssh_session.exec(shutdown_message(std::chrono::duration_cast<std::chrono::minutes>(delay)));
+
+        time_remaining = delay;
+        std::chrono::minutes time_elapsed{1};
+        QObject::connect(&shutdown_timer, &QTimer::timeout, [this, delay, time_elapsed]() mutable {
+            time_remaining = delay - time_elapsed;
+            ssh_session.exec(shutdown_message(std::chrono::duration_cast<std::chrono::minutes>(time_remaining)));
```

Comment:
Wonder if we should not spam every minute? Maybe once when first scheduled, then at 5 minutes, only then every minute?

Reply:
The `shutdown` command itself sends a message every minute when the time is set to 15 minutes or less.  Any time over that and you don't see a message at all, even when the time remaining dips below 15 minutes (which seems strange).

But we could be more nice on times greater than 5 minutes as not to peeve the user:)

Reply:
Yeah we don't have to be *that* inspired by `shutdown` ;)

Reply:
This should be better now.